### PR TITLE
Add cloud name for customizing specific Azure configuration

### PR DESCRIPTION
**Pull request by ms-hujia**
Add a new entry of Azure cloud name in azsettings, which is intended to represent specific Azure cloud in Grafana whose configuration is variable and will be customized by cloud admin or automated scripts.

### CODE REVIEW

**Comment by aangelisc** [comment]
Hi @ms-hujia, thanks for this! Would you be able to provide some more context for us on this parameter and what the use case is here? It'd be really appreciated 😊

**Comment by ms-hujia** [comment]
Hi @aangelisc, this PR is prerequisite of another PR in which I will make some related changes to the backend of Grafana Azure Monitor data-source. The initial idea to add this cloud name is to use Grafana in specific non-public Azure Cloud such as [Azure Stack Hub](https://azure.microsoft.com/en-us/products/azure-stack/hub/). Unlike other pre-defined clouds, they have variable configurations for authentication and service endpoints for different instances. So, it's not possible to hardcode those configurations in code or generate by rules. This cloud name works as an indicator to tell the Azure Monitor plugin that the authentication or service endpoints to be used will be provided with extra effort by admin or automated scripts.

**Review by andresmgot — COMMENTED** [review]
Hi @ms-hujia, thanks for the PR! I have some minor sugestions here.

In any case, it would be useful to know about all the changes before we release any changes. It could be misleading to see here that Azure on premise is added if it's not supported yet.

I think that you can work around this change in `grafana/grafana` and hardcode the name (for the moment) of the custom cloud in the plugin code so we can see all the changes and decide which code should go to the SDK.

The cloud handling is at the moment a bit spread over the plugin code and this SDK so it may make sense to refactor more code if we are modifying it.

**Comment by ms-hujia** [comment]
Hi @andresmgot, your suggestion makes sense. I will create the PR for my main changes in Grafana first.

File: azsettings/clouds.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ const (
 	AzureChina        = "AzureChinaCloud"
 	AzureUSGovernment = "AzureUSGovernment"
 	AzureGermany      = "AzureGermanCloud"
+	AzureCustomized   = "AzureCustomizedCloud"
```
[the remark this answers is not in the document]
Comment (ms-hujia) [reply azsettings/clouds.go:10]: This change should be common enough, not limited to on-premises cloud. Any non-public Azure cloud which is not given in the list can be configured using this option.

File: azsettings/clouds.go [hunk 2]
```diff
@@ -48,6 +49,16 @@ func NormalizeAzureCloud(cloudName string) string {
...
 		return AzureGermany
+
+	// Customized
+	case "azurecustomizedcloud":
+		fallthrough
+	case "azurecustomized":
+		fallthrough
+	case "custom":
+		fallthrough
+	case "customized":
```
[the remark this answers is not in the document]
Comment (ms-hujia) [reply azsettings/clouds.go:60]: No default case here, I'm just trying to align with the code convention/style around.

File: azsettings/clouds.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andresmgot) [reply azsettings/clouds.go:10]: fair point

File: azsettings/clouds.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (andresmgot) [reply azsettings/clouds.go:60]: okay, in any case I think we should simplify this and avoid unnecessary clutter. We can leave it as just:

```
	case "azurecustomizedcloud":
		return AzureCustomized
	}
```

**Review by andresmgot — COMMENTED** [review]
Thanks @ms-hujia for sending the complete PR in the other repo! That helps seeing the whole picture. I think we can move ahead with this PR once the remaining suggestion is addressed.

**Review by andresmgot — APPROVED** [review]
LGTM, thanks!

**Pull request merged**